pkg/beacon/relay/dkg/dkg.go

```diff
@@ -19,6 +20,8 @@ import (
 
 var logger = log.Logger("keep-dkg")
 
+const publicationDelayBlocks = 20
```

Comment:
I'd prefer having this in the state machine for `gjkr` as we are executing calculations on `gjkr` coproducts. And the value of this delay reflects the complexity of those calculations.

A possible idea how to get some of this code out of `dkg.go`:

In `states.go`, at `Initiate` of `combinationState` we can do:
```
cs.member.CombineGroupPublicKey()
cs.member.ComputeGroupPublicKeyShares()
```

`ComputeGroupPublicKeyShares` triggers a goroutine computing the shares asynchronously. Once the result is ready, the goroutine unblocks a `Cond` in `gjkr.Result`:

```
package gjkr

type Result struct {
  // (...)
  
  groupPublicKeySharesReady sync.Cond
  groupPublicKeyShares map[group.MemberIndex]*bn256.G2
}

func (r *Result) GroupPublicKeyShares() map[group.MemberIndex]*bn256.G2 {
  // wait for cond to be fulfilled and return groupPublicKeyShares
}
```

Then, in `dkg.go` we:
- get rid of `publicationDelayBlocks` and just increase the time for `combinationState` 
- proceed as before, remove.`groupPublicKeySharesChan` construction
- when we return the result we do `gjkrResult.GroupPublicKeyShares()` and this is possibly blocking

I think it's a little more elegant as the calculation of the shares is not leaking out of `gjkr`- we just may need to wait for `GroupPublicKeyShares()`. I was also thinking about using a channel instead of `sync.Cond` but this is problematic as the value can be read only once.

Reply:
Excellent idea but there is one problem. We must pass the result of `ComputeGroupPublicKeyShares` which works in a separate goroutine to `gjkr.Result` so we can't avoid an intermediate channel. Also `sync.Cond` is for cases when we have multiple goroutines waiting for an async result. In our case, a simple channel seems to be the best option. Apart from that, I agree we should push those computations inside GJKR. I will propose a solution inspired by this comment but without `sync.Cond` usage.